app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue

```diff
@@ -0,0 +1,198 @@
+import FilterInputBox from '../../../../components/widgets/FilterInput.vue';
+import countries from '/app/javascript/shared/constants/countries.js';
+import { mapGetters } from 'vuex';
+
+export default {
+  components: {
+    FilterInputBox,
+  },
+  mixins: [alertMixin],
+  props: {
+    onClose: {
+      type: Function,
+      default: () => {},
+    },
+    filterTypes: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  validations: {
+    appliedFilters: {
+      required,
+      $each: {
+        values: {
+          required,
+        },
+      },
+    },
+  },
+  data() {
+    return {
+      show: true,
+      appliedFilters: [],
+    };
+  },
+  computed: {
+    filterAttributes() {
+      return this.filterTypes.map(type => {
+        return {
+          key: type.attributeKey,
+          name: this.$t(`CONTACTS_FILTER.ATTRIBUTES.${type.attributeI18nKey}`),
+        };
+      });
+    },
+    ...mapGetters({
+      getAppliedContactFilters: 'contacts/getAppliedContactFilters',
+    }),
+    hasAppliedFilters() {
+      return this.getAppliedContactFilters.length;
+    },
+  },
+  mounted() {
+    if (this.getAppliedContactFilters.length) {
+      this.appliedFilters = this.getAppliedContactFilters;
+    } else {
+      this.appliedFilters.push({
+        attribute_key: 'name',
+        filter_operator: 'equal_to',
+        values: '',
+        query_operator: 'and',
+      });
+    }
+  },
+  methods: {
+    getInputType(key) {
+      const type = this.filterTypes.find(filter => filter.attributeKey === key);
+      return type.inputType;
+    },
+    getOperators(key) {
+      const type = this.filterTypes.find(filter => filter.attributeKey === key);
+      return type.filterOperators;
+    },
+    // eslint-disable-next-line consistent-return
+    getDropdownValues(type) {
+      switch (type) {
+        case 'country_code':
+          return countries;
+        default:
+          break;
+      }
+    },
+    appendNewFilter() {
+      this.appliedFilters.push({
+        attribute_key: 'name',
+        filter_operator: 'equal_to',
+        values: '',
+        query_operator: 'and',
+      });
+    },
+    removeFilter(index) {
+      if (this.appliedFilters.length <= 1) {
+        this.showAlert(this.$t('CONTACTS_FILTER.FILTER_DELETE_ERROR'));
+      } else {
+        this.appliedFilters.splice(index, 1);
+      }
+    },
+    submitFilterQuery() {
+      this.$v.$touch();
+      if (this.$v.$invalid) return;
+      this.appliedFilters[this.appliedFilters.length - 1].query_operator = null;
+      this.$store.dispatch(
+        'contacts/setContactFilters',
+        JSON.parse(JSON.stringify(this.appliedFilters))
+      );
+      this.$emit('applyFilter', this.appliedFilters);
+    },
+    resetFilter(index, currentFilter) {
+      this.appliedFilters[index].filter_operator = this.filterTypes.find(
+        filter => filter.attributeKey === currentFilter.attribute_key
```

Comment:
One object has camel key and other has snake key. We could make them consistent

Reply:
This snake case convention being is used in the `/api/v1/accounts/1/contacts/filter` api, this key is being used as an identifier to reset the filter operator value, which is linked to the type of filter selected above.

Same goes with the `/conversation/filter` endpoint